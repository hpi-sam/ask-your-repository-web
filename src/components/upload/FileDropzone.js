// @flow
import React from 'react';
import Dropzone from 'react-dropzone';
import type { DropFilesEventHandler } from 'react-dropzone';
import './FileDropzone.scss';

type Props = {
  onDrop: DropFilesEventHandler,
};

const FileDropzone = ({ onDrop }: Props) => (
  <Dropzone
    className="FileDropzone"
    activeClassName="FileDropzone--active"
    acceptClassName="FileDropzone--accept"
    disabledClassName="FileDropzone--disabled"
    onDrop={onDrop}
    data-cy="upload-dropzone"
  >
    <span className="FileDropzone__text">
      Place your file(s) here
    </span>
  </Dropzone>
);

export default FileDropzone;
