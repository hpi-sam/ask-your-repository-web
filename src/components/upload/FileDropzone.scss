@import '../../style/breakpoints.scss';

.FileDropzone {
  --bg-color: transparent;
  --hover-bg-color: rgba(255, 255, 255, .03);

  --border-width: 4px;
  --border-color: #fff;
  --hover-border-color: rgb(36, 186, 255);

  --text-color: #fff;
  --hover-text-color: rgb(36, 186, 255);

  --transition-duration: 0.2s;

  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  height: 100%;
  width: 100%;
  padding: 20px;

  background-color: var(--bg-color);
  border: var(--border-width) dashed var(--border-color);
  border-radius: inherit;

  transition:
    background-color var(--transition-duration),
    border-color var(--transition-duration),
    color var(--transition-duration);

  &:hover {
    --bg-color: var(--hover-bg-color);
    --text-color: var(--hover-text-color);
    --border-color: var(--hover-border-color);
  }

  &__text {
    text-align: center;
    font-weight: bold;
    letter-spacing: 3px;
    text-transform: uppercase;
    color: var(--text-color);
  }
}
