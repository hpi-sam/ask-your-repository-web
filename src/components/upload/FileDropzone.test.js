// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileDropzone from './FileDropzone';

describe('<FileDropzone />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FileDropzone onDrop={() => {}} />);
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
