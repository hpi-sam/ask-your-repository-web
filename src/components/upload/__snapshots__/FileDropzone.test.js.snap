// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileDropzone /> renders correctly 1`] = `
<n
  acceptClassName="FileDropzone--accept"
  activeClassName="FileDropzone--active"
  className="FileDropzone"
  disableClick={false}
  disabled={false}
  disabledClassName="FileDropzone--disabled"
  getDataTransferItems={[Function]}
  inputProps={Object {}}
  maxSize={Infinity}
  minSize={0}
  multiple={false}
  onDrop={[Function]}
  preventDropOnDocument={true}
>
  <span
    className="FileDropzone__text"
  >
    Place your file here
  </span>
</n>
`;
